### PR TITLE
[WIP] Introduce acceptance make target to run the tests

### PR DESCRIPTION
**Pull request by pmacik**
**What this PR does / why we need it**:

Introduce `acceptance` make target to run the tests with the possibility to switch between built and provided binary (via `BINDIR` variable)

**Special notes for your reviewer**:
-

**If applicable**:
- [ ] this PR contains documentation
- [ ] this PR contains unit tests
- [ ] this PR has been tested for backwards compatibility

### CODE REVIEW

**Comment by sbose78** [comment]
@akashshinde @pedjak @bamachrn

**Comment by pedjak** [comment]
@pmacik what is the motivation for this PR? You modified openshift-ci Makefile, hence I assume that CI will never have an opportunity to (re)use existing Helm binary?

File: openshift-ci/Makefile [hunk 1]
```diff
@@ -12,6 +12,16 @@ export ACCEPTANCE_RUN_TESTS := repos.robot
 export ROBOT_HELM_V3 := 1
 export ROBOT_DEBUG_LEVEL := 3
 
+ifdef BINDIR
```
Comment (pedjak) [inline openshift-ci/Makefile]: since we include the main `Makefile` in the line 3, `BINDIR` is defined already.

**Comment by pmacik** [comment]
> @pmacik what is the motivation for this PR? You modified openshift-ci Makefile, hence I assume that CI will never have an opportunity to (re)use existing Helm binary?

@pedjak The motivation is for the openshift-ci Makefile to be able to use either the "local" build of helm (that is built by `make build`) as it is the current state with the original `test-acceptance` target or to "skip" the build and run the acceptance tests with a helm binary that was built outside the frame of this repo - e.g. built by brew - to be able to verify that binary.

**Comment by pedjak** [comment]
In that case, I would not put conditional logic in `Makefile`, IMHO it is easier/more readable to put in a script and invoke it from `Dockerfile.tests`, something like this:

```
if [ -z "$1" ]; then
  make build
  HELM_PATH=$(pwd)/bin
else
  HELM_PATH=$1
fi

ROBOT_HELM_PATH=$HELM_PATH make -f acceptance-tests/Makefile acceptance
```

File: openshift-ci/.gitignore [hunk 2]
```diff
@@ -1,2 +1,3 @@
 .gocache
-.xdg-cache
\ No newline at end of file
+.xdg-cache
+check-patch
```
Comment (pedjak) [inline openshift-ci/.gitignore]: where this file comes from?

File: openshift-ci/Makefile [hunk 3]
```diff
@@ -23,3 +19,12 @@ lint: $(GOLANGCI_LINT) test-style
 
 .PHONY: unit
 unit: test-unit
+
+.PHONY: acceptance
+acceptance:
+	CURDIR=$(CURDIR) $(OPENSHIFT_CI)/acceptance-tests.sh
+
+.PHONY: clean-cache
+clean-cache:
```
Comment (pedjak) [inline openshift-ci/Makefile:28]: when one should use this rule?

File: openshift-ci/Makefile [hunk 4]
```diff
@@ -23,3 +19,12 @@ lint: $(GOLANGCI_LINT) test-style
 
 .PHONY: unit
 unit: test-unit
+
+.PHONY: acceptance
+acceptance:
```
Comment (pedjak) [inline openshift-ci/Makefile:24]: what is the benefit of calling acceptance-test script from Makefile?

**Pull request closed without merging**